### PR TITLE
don't build the F90 probdata_module

### PR DESCRIPTION
**Pull request by zingale**
we no longer need the Fortran runtime parameters

## PR checklist

- [ ] test suite needs to be run on this PR
- [ ] this PR will change answers in the test suite to more than roundoff level
- [ ] all newly-added functions have docstrings as per the coding conventions
- [ ] the `CHANGES` file has been updated, if appropriate
- [ ] if appropriate, this change is described in the docs

### CODE REVIEW

**Review by maxpkatz — APPROVED** [review]

**Comment by zingale** [comment]
a lot of failures: http://groot.astro.sunysb.edu/Castro/test-suite/gfortran/2021-05-31-002/index.html

**Comment by zingale** [comment]
tests: http://groot.astro.sunysb.edu/Castro/test-suite/gfortran/2021-05-31-004/index.html

**Comment by zingale** [comment]
all tests now pass: http://groot.astro.sunysb.edu/Castro/test-suite/gfortran/2021-06-01-004/index.html

File: Exec/Make.auto_source [hunk 1]
```diff
@@ -39,27 +39,27 @@ $(CASTRO_AUTO_SOURCE_DIR)/extern.F90: $(EXTERN_PARAMETERS) $(EXTERN_TEMPLATE)
...
 # search Bdirs for _prob_params files
 PROB_PARAMETERS := $(foreach dir, $(Bdirs), $(wildcard $(TOP)/$(dir)/_prob_params))
 # the problem's own parameters always come at the end so they override
 # any others
 PROB_PARAMETERS += $(wildcard $(PROBLEM_DIR)/_prob_params)
 
+$(CASTRO_AUTO_SOURCE_DIR)/prob_parameters.cpp: $(CASTRO_AUTO_SOURCE_DIR)/prob_parameters.H
+
+# for dependency resolution
+$(depEXETempDir)/%.d: $(CASTRO_AUTO_SOURCE_DIR)/prob_parameters.H
```
Comment (maxpkatz) [inline Exec/Make.auto_source:54]: Why is this needed?

Comment (zingale) [reply Exec/Make.auto_source:54]: we have to force the build system to create the header file before it does the C++ dependency resolution, otherwise it will fail, because it can't find the prob_parameters.H.

**Review by maxpkatz — APPROVED** [review]
